lib/tapyrus/bip175.rb

```diff
@@ -0,0 +1,66 @@
+module Tapyrus
+  # Key generation based on BIP-175
+  #
+  # @example
+  #
+  # master = Tapyrus::ExtKey.from_base58('xprv9s21ZrQH143K2JF8RafpqtKiTbsbaxEeUaMnNHsm5o6wCW3z8ySyH4UxFVSfZ8n7ESu7fgir8imbZKLYVBxFPND1pniTZ81vKfd45EHKX73')
+  # bip175 = Tapyrus::BIP175.from_private_key(master)
+  # bip175 << "foo"
+  # bip175 << "bar"
+  # bip175.addr
+  # > 1C7f322izqMqLzZzfzkPAjxBzprxDi47Yf
+  #
+  # @see https://github.com/bitcoin/bips/blob/master/bip-0175.mediawiki
+  class BIP175
+    PURPOSE_TYPE = 175
+
+    attr_accessor :payment_base
+
+    def initialize
+      @contracts = []
+    end
+
+    # @param key [Tapyrus::ExtKey] master private extended key
+    def self.from_ext_key(key)
```

Comment:
If you provide this method, you need to verify that the argument key is the master private extended key.
If it is not the master private extended key, you will get the wrong key derivation.

Reply:
It is good that ExtKey itself has the judgment of whether it is a master or not.